File: src/main/java/com/squarepolka/readyci/tasks/maven/JavaDependenciesMaven.java
package com.squarepolka.readyci.tasks.maven;

import com.squarepolka.readyci.taskrunner.BuildEnvironment;
import com.squarepolka.readyci.taskrunner.TaskFailedException;
import com.squarepolka.readyci.tasks.Task;
import org.springframework.stereotype.Component;

@Component
public class JavaDependenciesMaven extends Task {

    public static final String TASK_MAVEN_BUILD = "maven_install";

    @Override
    public String taskIdentifier() {
        return TASK_MAVEN_BUILD;
    }

    @Override
    public void performTask(BuildEnvironment buildEnvironment) throws TaskFailedException {
        executeCommand(new String[] {"mvn", "install"}, buildEnvironment.getProjectPath());
    }
}

File: src/main/kotlin/tasks/maven/JavaDependenciesMaven.kt
package tasks.maven

import com.squarepolka.readyci.taskrunner.BuildEnvironment
import com.squarepolka.readyci.taskrunner.TaskFailedException
import com.squarepolka.readyci.tasks.Task
import org.springframework.stereotype.Component

@Component
class JavaDependenciesMaven: Task() {

    val TASK_MAVEN_BUILD = "maven_install"

    override fun taskIdentifier(): String {
        return TASK_MAVEN_BUILD
    }

    @Throws(TaskFailedException::class)
    override fun performTask(buildEnvironment: BuildEnvironment) {
        executeCommand(arrayOf("mvn", "install"), buildEnvironment.projectPath)
    }

}